src/components/Message/Message.js

```diff
@@ -1,528 +1,251 @@
+      onUserHoverHandler: propOnUserHover,
+    },
+    message,
+  );
+  const isMuted = isUserMuted(message, mutes);
+  const { isMyMessage, isAdmin, isModerator, isOwner } = useUserRole(message);
+  const canEdit = isMyMessage || isModerator || isOwner || isAdmin;
+  const canDelete = canEdit;
+  const messageActionsHandler = useCallback(() => {
     if (!message || !messageActions) {
       return [];
     }
 
     return getMessageActions(messageActions, {
-      canDelete: this.canDeleteMessage(message),
-      canEdit: this.canEditMessage(message),
-      canFlag: !this.isMyMessage(message),
-      canMute: !this.isMyMessage(message) && !!channelConfig?.mutes,
+      canDelete,
+      canEdit,
+      canFlag: !isMyMessage,
+      canMute: !isMyMessage && !!channelConfig?.mutes,
     });
-  };
-
-  render() {
-    const { channel, message } = this.props;
-    const config = channel?.getConfig && channel.getConfig();
-
-    const actionsEnabled =
-      message && message.type === 'regular' && message.status === 'received';
-
-    const MessageUIComponent = this.props.Message;
-    return (
-      MessageUIComponent && (
-        <MessageUIComponent
-          {...this.props}
-          actionsEnabled={actionsEnabled}
-          Message={MessageUIComponent}
-          handleReaction={this.handleReaction}
-          getMessageActions={this.getMessageActions}
-          handleFlag={this.handleFlag}
-          handleMute={this.handleMute}
-          handleAction={this.handleAction}
-          handleDelete={this.handleDelete}
-          handleEdit={this.handleEdit}
-          handleRetry={this.handleRetry}
-          handleOpenThread={
-            message &&
-            this.props?.openThread &&
-            this.props.openThread.bind(this, message)
-          }
-          isUserMuted={this.isUserMuted}
-          isMyMessage={this.isMyMessage}
-          channelConfig={config}
-          onMentionsClickMessage={this.onMentionsClick}
-          onMentionsHoverMessage={this.onMentionsHover}
-          onUserClick={this.onUserClick}
-          onUserHover={this.onUserHover}
-        />
-      )
-    );
-  }
-}
-
-export default withTranslationContext(Message);
+  }, [channelConfig, message, messageActions, canDelete, canEdit, isMyMessage]);
+
+  const actionsEnabled =
+    message && message.type === 'regular' && message.status === 'received';
+
+  return (
+    MessageUIComponent && (
+      <MessageUIComponent
+        {...props}
+        actionsEnabled={actionsEnabled}
+        Message={MessageUIComponent}
+        handleReaction={handleReaction}
+        getMessageActions={messageActionsHandler}
+        handleFlag={handleFlag}
+        handleMute={handleMute}
+        handleAction={handleAction}
+        handleDelete={handleDelete}
+        handleEdit={handleEdit}
+        handleRetry={handleRetry}
+        handleOpenThread={handleOpenThread}
+        isUserMuted={() => isMuted}
+        isMyMessage={() => isMyMessage}
```

Comment:
this can potentially be a breaking change to just return the value instead of the wrapper fn, can you add a deprecated note for the next major release.